substratest/assets.py

```diff
@@ -50,6 +50,50 @@ def get(self):
         return self._asset
 
 
+class ComputePlanFuture(Future):
+    _keys_properties = {
+        'ComputePlan': {
+            'traintuple_keys': 'traintuples',
+            'composite_traintuple_keys': 'composite_traintuples',
+            'aggregatetuple_keys': 'aggregatetuples',
+            'testtuple_keys': 'testtuples',
+        },
+        'ComputePlanCreated': {
+            'traintuple_keys': 'traintuple_keys',
+            'composite_traintuple_keys': 'composite_traintuple_keys',
+            'aggregatetuple_keys': 'aggregatetuple_keys',
+            'testtuple_keys': 'testtuple_keys',
+
+        },
+    }
```

Comment:
Even if it takes more line, I think setting these properties manually would by easier to read and to maintain. 

We could handle the different case in the constructor. For instance:
```python
self.traintuples_keys = asset.traintuples if hasattr(asset, 'traintuples') else asset.traintuple_keys 
```

One other solution would be to add the required properties to the `ComputePlan` class (`traintuple_keys`...).

What do you think?

Reply:
As discussed in the office, I went with the last solution.